app/controllers/planning_applications_controller.rb

```diff
@@ -58,12 +49,9 @@ def update
 
   def assign
     if request.patch?
-      @planning_application.user = if params[:planning_application][:user_id] == "0"
-                                     nil
-                                   else
-                                     current_local_authority.users.find(params[:planning_application][:user_id])
-                                   end
-      audit("assigned", nil, @planning_application.user&.name)
+      user = params[:planning_application][:user_id].eql?("0") ? nil : User.find(params[:planning_application][:user_id])
+      @planning_application.assign(user)
```

Comment:
Do we still need to scope users to `current_local_authority.users` ?
And a bit confused with the logic here of setting user to nil if the params is 0 🤔 
I mean will there ever be a user with id of 0

Reply:
see - https://github.com/unboxed/bops/pull/508#discussion_r768660347

Reply:
in the view: app/views/planning_applications/assign.html.erb

![image](https://user-images.githubusercontent.com/9452321/146025943-ea99a567-99c0-40f4-adde-3fa6d1b155a4.png)

Reply:
Hm I think we need to scope it to be safe. Because it's using params there is ability for a bad actor to tamper with it